code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -396,6 +411,11 @@
 	hat_offset = -5
 	clean_on_move = TRUE
 
+/obj/item/robot_module/janitor/do_transform_animation()
+	var/mob/living/silicon/robot/R = loc
+	flick("janiborg_transform", R)

```

Comment:
to be frank, you shouldn't have to have to give each subtype a proc with this flick; it should be handled with a var, `transform_icon`, that's checked to see if it has a transformation icon and then, if it doesn't, does the old behavior instead.

iiiit's also super crude that some borgs will appear to be visually transformed but still unable to move because the transformation isn't over.

Reply:
To be fair, All I did was snag the code you had for peacekeepers and changed the flick key. I won't exaggerate my ability to fix a lot of this when it's not done in any other code that I can reference easily.

unless something like `"[R.module]+"_transform",R)` will work

Reply:
flick("[cyborg_base_icon]_transform", R) would work, if you renamed the transform animations to line up with that var.

Reply:
Aye, that I did, and it works. yey